### ISSUE TITLE
Bug: status for dependency validation should be changed to 500.

### ISSUE THREAD

**Issue by peterschutt** [starlite-api/starlite#194]
Originally from #138

When a value returned from a provided dependency fails pydantic validation, user gets a 4xx, but it should really be a 5xx.

Ref: https://github.com/starlite-api/starlite/issues/138#issuecomment-1157066885_

**Comment by peterschutt** [comment]
Any advice on process for this one @Goldziher?

**Comment by Goldziher** [comment]
Basically there is one try/except block that handles this: https://github.com/starlite-api/starlite/blob/main/starlite/signature.py#L56

You can probably add an attribute on the `SignatueModel` that will tell you this info, see the attributes already there - which use a `ClassVar` so pydantic ignores them: https://github.com/starlite-api/starlite/blob/main/starlite/signature.py#L21

**Comment by peterschutt** [comment]
Yes OK - so perhaps a set of the names of all function parameters that are known dependencies as a classvar on the sig model and inspect that after the validation error to determine if 4xx or 5xx?

**Comment by Goldziher** [comment]
Thats how i'd approach it
